specification/http/1.0/openapi.json

```diff
@@ -0,0 +1,469 @@
+      "get": {
+        "operationId": "list-tools",
+        "summary": "List tools",
+        "description": "Returns a list of tool definitions.",
+        "security": [
+          {
+            "bearerAuth": []
+          },
+          {}
+        ],
+        "responses": {
+          "200": {
+            "description": "A list of available tools.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "properties": {
+                    "$schema": {
+                      "type": "string",
+                      "format": "uri"
+                    },
+                    "tools": {
+                      "type": "array",
+                      "items": {
+                        "$ref": "#/components/schemas/ToolDefinition"
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/tools/execute": {
+      "post": {
+        "operationId": "execute-tool",
+        "summary": "Execute tool",
+        "description": "Executes a tool with the given parameters.",
+        "security": [
+          {
+            "bearerAuth": []
+          },
+          {}
+        ],
+        "requestBody": {
+          "required": true,
+          "content": {
+            "application/json": {
+              "schema": {
+                "properties": {
+                  "$schema": {
+                    "type": "string",
+                    "format": "uri"
+                  },
+                  "request": {
+                    "$ref": "#/components/schemas/ExecuteToolRequest"
+                  }
+                },
+                "required": ["request"]
+              }
+            }
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "Execution response from tool.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "properties": {
+                    "$schema": {
+                      "type": "string",
+                      "format": "uri"
+                    },
+                    "response": {
+                      "$ref": "#/components/schemas/ExecuteToolResponse"
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "400": {
+            "description": "The request is invalid."
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "securitySchemes": {
+      "bearerAuth": {
+        "type": "http",
+        "scheme": "bearer",
+        "bearerFormat": "JWT"
+      }
+    },
+    "schemas": {
+      "ToolId": {
+        "type": "string",
+        "description": "A tool's unique identifier in the format 'Toolkit.Tool[@version]', where @version is optional.",
+        "pattern": "^[A-Za-z0-9_]+\\.[A-Za-z0-9_]+(@[A-Za-z0-9_.\\-]+)?$",
+        "example": [
+          "MyToolkit.MyTool",
```

Comment:
Is it useful to allow for `/` in the toolkit name for namespacing purposes? Or do we want to handle?

google/gmail
google/calendar

---

What are semantics of `Calculator.Add@latest` if Add was present in version 1, but removed in version 2. Version 2 is the latest.